src/base/bittorrent/infohash.h

```diff
@@ -39,7 +45,11 @@ namespace BitTorrent
     {
     public:
         InfoHash();
+#if LIBTORRENT_VERSION_NUM < 20000
         InfoHash(const lt::sha1_hash &nativeHash);
+#else
+        InfoHash(const lt::info_hash_t &nativeHash);
```

Comment:
If you define an alias for libtorrent infohash type, you will get fewer conditional expressions throughout the code.

```c++
#if LIBTORRENT_VERSION_NUM < 20000
using NativeHash = lt::sha1_hash;
#else
using NativeHash = lt::info_hash_t;
#endif
```

Reply:
Yes, this is true. I did not do use alias to be more "descriptive", but I'll change it now.

Reply:
Still, the majority of new LOC in this patch comes from having to choose between `info_hash` or `info_hashes` members. It's possible to alias `info_hashes` to `info_hash` for libtorrent 1.x, but that would be ugly, I think.